### PR TITLE
feat(Stateful Ingestion): 1/3  Stateful ingestion server changes

### PR DESCRIPTION
**Pull request by rslanka**
## Checklist
- [x ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-models/src/main/pegasus/com/linkedin/datajob/datahub/IngestionCheckpointState.pdl [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  /**
+   * The serialization/deserialization protocol.
+   */
+  serde: string
+
+  /**
+   * Opaque blob of the state representation.
+   */
+  byte_payload: optional bytes
```
Comment (shirshanka) [inline metadata-models/src/main/pegasus/com/linkedin/datajob/datahub/IngestionCheckpointState.pdl:24]: convert to camel case ... or drop the `byte_` prefix.

Comment (shirshanka) [reply metadata-models/src/main/pegasus/com/linkedin/datajob/datahub/IngestionCheckpointState.pdl:24]: camel-case option would be `payloadInBytes`

Comment (shirshanka) [reply metadata-models/src/main/pegasus/com/linkedin/datajob/datahub/IngestionCheckpointState.pdl:24]: I would recommend just `payload`

File: metadata-service/restli-client/src/main/java/com/linkedin/entity/client/JavaEntityClient.java [hunk 2]
```diff
@@ -52,342 +51,324 @@
 @Slf4j
 public class JavaEntityClient implements EntityClient {
 
-    private final Clock _clock = Clock.systemUTC();
```
Comment (shirshanka) [inline metadata-service/restli-client/src/main/java/com/linkedin/entity/client/JavaEntityClient.java:55]: seems like unnecessary formatting related changes

**Review by shirshanka — CHANGES_REQUESTED** [review]
one show-stopper on field name. 

a few unnecessary formatting related changes on some files.. would be good to see if this can be avoided...

File: metadata-models/src/main/pegasus/com/linkedin/datajob/datahub/IngestionCheckpointState.pdl (hunk 1 above)
Comment (rslanka) [reply metadata-models/src/main/pegasus/com/linkedin/datajob/datahub/IngestionCheckpointState.pdl:24]: looked perfect on the python side, making the change ;-)

File: metadata-service/restli-client/src/main/java/com/linkedin/entity/client/JavaEntityClient.java (hunk 2 above)
Comment (rslanka) [reply metadata-service/restli-client/src/main/java/com/linkedin/entity/client/JavaEntityClient.java:55]: will fix.

File: metadata-io/src/main/java/com/linkedin/metadata/timeseries/transformer/TimeseriesAspectTransformer.java [hunk 3]
```diff
@@ -81,8 +81,17 @@ private static ObjectNode getCommonDocument(@Nonnull final Urn urn, final Record
         (Long) timeseriesAspect.data().get(MappingsBuilder.TIMESTAMP_MILLIS_FIELD));
     Object eventGranularity = timeseriesAspect.data().get(MappingsBuilder.EVENT_GRANULARITY);
     if (eventGranularity != null) {
-      document.put(MappingsBuilder.EVENT_GRANULARITY, eventGranularity.toString());
+      try {
+        document.put(MappingsBuilder.EVENT_GRANULARITY, OBJECT_MAPPER.writeValueAsString(eventGranularity));
+      } catch (JsonProcessingException e) {
+        throw new IllegalArgumentException("Failed to convert eventGranulairty to Json string!", e);
```
Comment (shirshanka) [inline metadata-io/src/main/java/com/linkedin/metadata/timeseries/transformer/TimeseriesAspectTransformer.java:87]: are transformers allowed to throw runtime exceptions?

**Review by shirshanka — APPROVED** [review]
LGTM

File: metadata-io/src/main/java/com/linkedin/metadata/timeseries/transformer/TimeseriesAspectTransformer.java (hunk 3 above)
Comment (shirshanka) [reply metadata-io/src/main/java/com/linkedin/metadata/timeseries/transformer/TimeseriesAspectTransformer.java:87]: is this change backwards compatible with the previous code?

**Review by shirshanka — APPROVED** [review]
LGTM
